### ISSUE TITLE
Disallow Certain Rules from Inline Error Supression

### ISSUE THREAD

**Issue by travis-cook-sfdc** [astral-sh/ruff#8797]
hi there -

I help to manage a fairly large repo at my company.

In certain scenarios, we'd like to use ruff rule [TID251](https://docs.astral.sh/ruff/rules/banned-api/) to help prevent certain functions from being used in our repo.

While this rule currently works fine, anyone can add `noqa: TID251` to their code.  I have no problem with someone doing this, but I wish there was a way to have this trigger a required review by the repo owner.

Ideally, we could use `per-file-ignores` to ignore this rule in certain allow-listed cases, but there's a few issues here:

```
[tool.ruff.per-file-ignores]
"__init__.py" = ["TID251"]
"path/to/file.py" = ["TID251"]
```

1️⃣  It's still possible to do an in-line ignore.  It could be nice to not allow inline ignores for this rule
2️⃣ This allows all banned apis in this file as opposed to just one specific.

Perhaps ruff isn't the right tool for the job here, but wanted to bring this up as a potential feature.

**Comment by KotlinIsland** [comment]
take a look at [pylint-module-boundaries](https://pypi.org/project/pylint-module-boundaries/), maybe that would suite your situation better
